proposal/indented-syntax-improvements.md

```diff
@@ -0,0 +1,215 @@
+meaningful, custom handling of line breaks is outside of the scope of this
+proposal. For instance, `@media (hover: hover) and \n (color)` is not supported,
+even though line breaks do not end statements after boolean operators in general
+SassScript.
+
+We considered borrowing alternate syntax from other languages, such as a leading
+`>` or `|` from YAML or a trailing `|` from HAML. These introduce syntax that is
+novel to Sass, and we instead opted to borrow syntax from the Scss format.
+
+These changes also allow authors using the indented syntax to add more explicit
+blocks with curly braces and line breaks with semicolons. While this introduces
+the changes for all authors, authors will still be able choose to limit the
+syntax with linters.
+
+In the indented syntax, authors can use a semicolon to explicitly end a
+statement. Subsequent lines in the same block still must have the same level of
+indentation. This means that the indented format won't support multiple
+statements on a single line, even if they are separated by a semicolon.
+
+```sass
+$font-stack: Helvetica, sans-serif;
+// or
+$primary-color: #333
+```
+
+In the indented syntax, authors can use curly braces to explicitly wrap a block
+of statements, but must use semicolons to separate statements inside that block.
+The braces essentially let authors opt in to Scss format for part of the
+document. We opted to require semicolons inside of curly braces to prevent
+issues with multiple nested formats.
+
+```sass
+$font-stack: Helvetica, sans-serif
+$primary-color: #333
+body {
+  font: 100% $font-stack;
+  color: $primary-color;
+}
+```
+
+When line breaks do not end statements, the line breaks are treated as white
+space, and do not have any requirements about the amount of indentation. The
+next statement's indentation is compared to the indentation at the start of the
+current statement to determine nesting or block closure.
+
+Line breaks that do not end statements are purely aesthetic and possible in a
+wide range of syntaxes, so it is up to the author to determine how to format
+their stylesheet.
+
+## Syntax
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement ';'? [IndentSame])\* Statement ';'?
+</pre></x>
+
+[IndentSame]: ../spec/statement.md#indentation
+
+Remove the following prose:
+
+The `Statement` productions may not include newlines outside of `IndentSame`
+productions.
+
+### Block
+
+Replace footnote 1 with:
+
+1: In the Scss syntax, only the `ScssBlock` production is valid.
+
+### WhitespaceComment
+
+In footnote 2, remove:
+
+In the indented syntax, this may not contain newlines.
+
+### Whitespace
+
+Replace footnote 1 with:
+
+1. In the indented syntax, `LineBreak` is not whitespace in the `IncludeAtRule`,
+   `SupportsAtRule`, [`MediaAtRule`], `KeyframesAtRule`, [`UnknownAtRule`], or
+   in any position where the semantics of a `Statement` define a possible end.
```

Comment:
@nex3 I'd welcome your feedback on if there is a better way to say "where the semantics of a `Statement` define a possible end."

I would like to say "When a linebreak is ambiguous as Whitespace or a PossibleStatementEnd, parse it preferentially as a PossibleStatementEnd", but the PossibleStatementEnd concept doesn't (and shouldn't) exist.  

Would it make sense to add syntax for Expression, roughly something like-

```
**Expression**  ::= Whitespace?¹  SingleExpression (WhiteSpace² SingleExpression)*
```
Existing: 
1 & 2: This may not include newlines in the indented syntax

Proposal:

1: This may include newlines in the indented syntax
2: This may not include newlines in the indented syntax.

`SingleExpression` could still contain newlines, so I'm unsure if this helps specify it, or if it just pushes the question of where newlines cause statement breaks another layer deeper.

Reply:
I'd lean on the same wording we're already using in [`ScssStatements`](java/com/google/assistant/aog/console/web/modules/languages/) and say something like "If a `LineBreak` in `Whitespace` would cause it to be ambiguous with an `IndentSame` or `IndentMore` production, parse it preferentially as `IndentSame` or `IndentMore`."